src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,175 @@
+ * This class provides methods to read data from a file, process it, and write the
+ * results to another file.
+ */
 public class WorkWithFile {
+    private static final String READ_DATA_EXCEPTION = "Can't read data from the file ";
+    private static final String WRITE_DATA_EXCEPTION = "Can't write data to the file ";
+    private static final String SUPPLY = "supply";
+    private static final String RESULT = "result,";
+    private static final String COMMA = ",";
+    private static final String SPACE = " ";
+    private static final int INDEX_ONE = 1;
+    private static final int INDEX_ZERO = 0;
+
+    /**
+     * Reads data from the specified file, processes the statistics, and writes the
+     * result to another file.
+     *
+     * @param fromFileName The name of the file to read data from.
+     * @param toFileName   The name of the file to write the result to.
+     */
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] content = readFromFile(fromFileName);
+        writeToFile(content, toFileName);
+    }
+
+    /**
+     * Reads content from the specified file and processes it to return an array
+     * containing statistics.
+     *
+     * @param fileName The name of the file to read data from.
+     * @return An array containing processed statistics.
+     * @throws RuntimeException If an error occurs while reading the file.
+     */
+    private String[] readFromFile(String fileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String value = reader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(SPACE);
+                value = reader.readLine();
+            }
+            return prepareStatistic(stringBuilder.toString().split(SPACE));
+        } catch (IOException e) {
+            throw new RuntimeException(READ_DATA_EXCEPTION + fileName, e);
+        }
+    }
+
+    /**
+     * Writes the given content to the specified file.
+     *
+     * @param content  The content to write to the file.
+     * @param fileName The name of the file to write data to.
+     * @throws RuntimeException If an error occurs while writing to the file.
+     */
+    public void writeToFile(String[] content, String fileName) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(fileName))) {
+            for (String line : content) {
+                writer.write(line);
+                writer.newLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(WRITE_DATA_EXCEPTION + fileName, e);
+        }
+    }
+
+    /**
+     * Processes the raw data array and returns an array containing statistics.
+     *
+     * @param array The raw data array to process.
+     * @return An array containing processed statistics.
+     */
+    private String[] prepareStatistic(String[] array) {
```

Comment:
This method is really big, but after a few hours of frustrating, it was the only solution I found to make sure that "supply" will be always in the first row...